### ISSUE TITLE
Unfolded JSON columns causing high load on DW during fingerprinting, which fails and continues forever

### ISSUE THREAD

**Issue by blevkovych** [metabase/metabase#25331]
**Describe the bug**
At certain times (23:00 UTC) Metabase queries will trigger CPU spikes on RDS instance.

**Logs**

<details><summary>Click to expand</summary>

```
2022-07-18 23:09:00,210 INFO task.sync-databases :: Starting sync task for Database 101.
2022-07-18 23:09:00,213 INFO sync.util :: [35mSTARTING: Sync metadata for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:00,214 INFO sync.util :: [35mSTARTING: step 'sync-timezone' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:00,233 INFO sync.util :: [35mFINISHED: step 'sync-timezone' for postgres Database 101 'analytics-test-pg' (19.8 ms)[0m
2022-07-18 23:09:00,234 INFO sync.util :: [35mSTARTING: step 'sync-tables' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:02,142 WARN keyverifier.AcceptAllServerKeyVerifier :: Server at /IP:22 presented unverified EC key: STERILIZED
2022-07-18 23:09:02,224 INFO sync.util :: [35mFINISHED: step 'sync-tables' for postgres Database 101 'analytics-test-pg' (2.0 s)[0m
2022-07-18 23:09:02,224 INFO sync.util :: [35mSTARTING: step 'sync-fields' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:31,209 INFO fields.sync-metadata :: Database type of Table 2,122 'fprdb_audit_logs.model_audits' Field 'new_data → address_confirmed' has changed from 'boolean' to 'integer'.
2022-07-18 23:09:31,209 INFO fields.sync-metadata :: Base type of Table 2,122 'fprdb_audit_logs.model_audits' Field 'new_data → address_confirmed' has changed from ':type/Boolean' to ':type/Integer'.
2022-07-18 23:09:31,233 INFO fields.sync-metadata :: Database type of Table 2,122 'fprdb_audit_logs.model_audits' Field 'changes → updated_at' has changed from 'timestamp' to 'text'.
2022-07-18 23:09:31,233 INFO fields.sync-metadata :: Base type of Table 2,122 'fprdb_audit_logs.model_audits' Field 'changes → updated_at' has changed from ':type/DateTime' to ':type/Text'.
2022-07-18 23:09:32,720 INFO sync.util :: [35mFINISHED: step 'sync-fields' for postgres Database 101 'analytics-test-pg' (30.5 s)[0m
2022-07-18 23:09:32,720 INFO sync.util :: [35mSTARTING: step 'sync-fks' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:39,399 INFO sync.util :: [35mFINISHED: step 'sync-fks' for postgres Database 101 'analytics-test-pg' (6.7 s)[0m
2022-07-18 23:09:39,400 INFO sync.util :: [35mSTARTING: step 'sync-metabase-metadata' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:41,405 INFO sync.util :: [35mFINISHED: step 'sync-metabase-metadata' for postgres Database 101 'analytics-test-pg' (2.0 s)[0m
2022-07-18 23:09:41,518 INFO sync.util :: [35mFINISHED: Sync metadata for postgres Database 101 'analytics-test-pg' (41.3 s)[0m
2022-07-18 23:09:41,518 INFO sync.util :: [35mSTARTING: Analyze data for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:41,538 INFO sync.util :: [35mSTARTING: step 'fingerprint-fields' for postgres Database 101 'analytics-test-pg'[0m
2022-07-18 23:09:41,975 INFO sync.analyze :: [34mfingerprint-fields Analyzed [*****·············································]    10% Table 1,964 'fprdb_vetiplan_api.vet_vet_meeting'[0m
2022-07-18 23:09:42,658 WARN sync.util :: Error fingerprinting Table 1,971 'fprdb_vetiplan_api.chat_messages'
clojure.lang.ExceptionInfo: Error executing query: ERROR: value "1618128596739" is out of range for type integer {:sql "-- Metabase\nSELECT \"source\".\"substring1158188\" AS \"substring1158188\", \"source\".\"read_at\" AS \"read_at\", \"source\".\"substring1158189\" AS \"substring1158189\", \"source\".\"_fivetran_deleted\" AS \"_fivetran_deleted\", \"source\".\"substring1158190\" AS \"substring1158190\", \"source\".\"substring1158191\" AS \"substring1158191\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → created_at\", \"source\".\"substring1158192\" AS \"substring1158192\", \"source\".\"substring1158193\" AS \"substring1158193\", \"source\".\"substring1158194\" AS \"substring1158194\", \"source\".\"sender_id\" AS \"sender_id\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → region_id\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → vet_id\", (\"source\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → permissions\", \"source\".\"substring1158195\" AS \"substring1158195\", \"source\".\"substring1158196\" AS \"substring1158196\", \"source\".\"substring1158197\" AS \"substring1158197\", \"source\".\"substring1158198\" AS \"substring1158198\", \"source\".\"substring1158199\" AS \"substring1158199\", \"source\".\"notified_at\" AS \"notified_at\", \"source\".\"substring1158200\" AS \"substring1158200\", \"source\".\"substring1158201\" AS \"substring1158201\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → user_id\", \"source\".\"receiver_id\" AS \"receiver_id\", \"source\".\"substring1158202\" AS \"substring1158202\", (\"source\".\"payload\"#>> ?::text[])::double precision  AS \"payload → data → sender → longitude\", \"source\".\"substring1158203\" AS \"substring1158203\", \"source\".\"substring1158204\" AS \"substring1158204\", (\"source\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → roles\", \"source\".\"substring1158205\" AS \"substring1158205\", (\"source\".\"payload\"#>> ?::text[])::text  AS \"payload → translations\", \"source\".\"substring1158206\" AS \"substring1158206\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → item → id\", \"source\".\"updated_at\" AS \"updated_at\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → id\", \"source\".\"_fivetran_synced\" AS \"_fivetran_synced\", \"source\".\"created_at\" AS \"created_at\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → animalId\", \"source\".\"substring1158207\" AS \"substring1158207\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → id\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → message_id\", (\"source\".\"payload\"#>> ?::text[])::double precision  AS \"payload → data → sender → latitude\", \"source\".\"substring1158208\" AS \"substring1158208\", (\"source\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → company_role → id\", \"source\".\"substring1158209\" AS \"substring1158209\", \"source\".\"substring1158210\" AS \"substring1158210\" FROM (SELECT \"fprdb_vetiplan_api\".\"chat_messages\".\"id\" AS \"id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → created_at\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → custom_type\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → animalId\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → cta_text\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → cta_url\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → description\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → item → id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → image → sizes → lg\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → image → sizes → md\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → image → sizes → sm\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → image → sizes → xl\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → image → url\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → opening_type\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → title\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → item → type\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → company_role → di_417dd22a\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → company_role → id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → company_role → name\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → email\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → first_name\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → last_name\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::double precision  AS \"payload → data → sender → latitude\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::double precision  AS \"payload → data → sender → longitude\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → name\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → permissions\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → phone\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → region_id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → roles\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → sender → vet_id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → sender → vet_uid\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → data → type\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → data → user_id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → message\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::integer  AS \"payload → message_id\", (\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text  AS \"payload → translations\", \"fprdb_vetiplan_api\".\"chat_messages\".\"read_at\" AS \"read_at\", \"fprdb_vetiplan_api\".\"chat_messages\".\"updated_at\" AS \"updated_at\", \"fprdb_vetiplan_api\".\"chat_messages\".\"notified_at\" AS \"notified_at\", \"fprdb_vetiplan_api\".\"chat_messages\".\"channel_url\" AS \"channel_url\", \"fprdb_vetiplan_api\".\"chat_messages\".\"sender_id\" AS \"sender_id\", \"fprdb_vetiplan_api\".\"chat_messages\".\"receiver_id\" AS \"receiver_id\", \"fprdb_vetiplan_api\".\"chat_messages\".\"payload\" AS \"payload\", \"fprdb_vetiplan_api\".\"chat_messages\".\"created_at\" AS \"created_at\", \"fprdb_vetiplan_api\".\"chat_messages\".\"_fivetran_deleted\" AS \"_fivetran_deleted\", \"fprdb_vetiplan_api\".\"chat_messages\".\"_fivetran_synced\" AS \"_fivetran_synced\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158192\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158195\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158208\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158203\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158191\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158204\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158202\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158207\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158193\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158196\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158198\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158194\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158197\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158205\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158188\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158201\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158189\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158209\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158200\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158199\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158190\", substring(\"fprdb_vetiplan_api\".\"chat_messages\".\"channel_url\", 1, 1234) AS \"substring1158206\", substring((\"fprdb_vetiplan_api\".\"chat_messages\".\"payload\"#>> ?::text[])::text , 1, 1234) AS \"substring1158210\" FROM \"fprdb_vetiplan_api\".\"chat_messages\") \"source\" LIMIT 10000", :params ("{created_at}" "{data,sender,region_id}" "{data,sender,vet_id}" "{data,sender,permissions}" "{data,user_id}" "{data,sender,longitude}" "{data,sender,roles}" "{translations}" "{data,item,id}" "{data,id}" "{data,animalId}" "{data,sender,id}" "{message_id}" "{data,sender,latitude}" "{data,sender,company_role,id}" "{created_at}" "{custom_type}" "{data,animalId}" "{data,id}" "{data,item,cta_text}" "{data,item,cta_url}" "{data,item,description}" "{data,item,id}" "{data,item,image,sizes,lg}" "{data,item,image,sizes,md}" "{data,item,image,sizes,sm}" "{data,item,image,sizes,xl}" "{data,item,image,url}" "{data,item,opening_type}" "{data,item,title}" "{data,item,type}" "{data,sender,company_role,display_name}" "{data,sender,company_role,id}" "{data,sender,company_role,name}" "{data,sender,email}" "{data,sender,first_name}" "{data,sender,id}" "{data,sender,last_name}" "{data,sender,latitude}" "{data,sender,longitude}" "{data,sender,name}" "{data,sender,permissions}" "{data,sender,phone}" "{data,sender,region_id}" "{data,sender,roles}" "{data,sender,vet_id}" "{data,sender,vet_uid}" "{data,type}" "{data,user_id}" "{message}" "{message_id}" "{translations}" "{data,type}" "{data,item,description}" "{data,item,opening_type}" "{message}" "{data,item,cta_url}" "{data,item,cta_text}" "{data,sender,email}" "{data,item,image,sizes,sm}" "{data,sender,vet_uid}" "{data,item,image,url}" "{data,sender,first_name}" "{data,sender,phone}" "{data,sender,name}" "{data,sender,company_role,display_name}" "{custom_type}" "{data,sender,company_role,name}" "{data,item,image,sizes,lg}" "{data,item,title}" "{data,item,image,sizes,xl}" "{data,item,image,sizes,md}" "{data,sender,last_name}" "{data,item,type}"), :type :invalid-query}
	at metabase.driver.sql_jdbc.execute$execute_reducible_query$fn__54060.invoke(execute.clj:502)
	at metabase.driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:499)
	at metabase.driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:486)
```

</details>

Problematic query:
`SELECT ("fprdb_audit_logs"."model_audits"."new_data"#>> $1::text[])::bigint  AS "new_data → pinned_by_id" FROM "fprdb_audit_logs"."model_audits" GROUP BY "new_data → pinned_by_id" ORDER BY "new_data → pinned_by_id" ASC LIMIT ?`

**To Reproduce**

- Use Metabase

**Expected behavior**

- No CPU spikes on my RDS instance.

**Screenshots**
![Screenshot 2022-09-09 at 16 50 00](https://user-images.githubusercontent.com/38095999/189378654-065488b6-dbf2-4949-bc55-c25c1b810460.png)
![Screenshot 2022-09-09 at 16 51 51](https://user-images.githubusercontent.com/38095999/189383346-239c4dc1-6fbf-4a0b-b255-90f14267cfc3.png)

**Information about your Metabase Installation:**

`{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.16+8-post-Ubuntu-0ubuntu118.04",
    "java.vendor": "Ubuntu",
    "java.vendor.url": "https://ubuntu.com/",
    "java.version": "11.0.16",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.16+8-post-Ubuntu-0ubuntu118.04",
    "os.name": "Linux",
    "os.version": "4.15.0-176-generic",
    "user.language": "en",
    "user.timezone": "Europe/Berlin"
  },
  "metabase-info": {
    "databases": [
      "googleanalytics",
      "postgres",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.4"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.4.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-08-31",
      "tag": "v0.44.2",
      "branch": "release-x.44.x",
      "hash": "8d9de8a"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}`

- Your browser and the version: Not relevant
- Your operating system: Not relevant
- Your databases: PostgreSQL
- Metabase version: v0.44.2
- Metabase hosting environment: DigitalOcean Droplet 
- Metabase internal database: PostgreSQL

**Severity**
Right now it's just scary, but may kill the database.

**Additional context**
Started from 08.25.2022 v0.43.4 and continues with v0.44.2, didn't happen before.

**Comment by flamber** [comment]
This is basically the same as #19470, but just specific to unfolded JSON columns being fingerprinted and failing, so it will just continue to do this forever.

You can disable scanning: https://www.metabase.com/docs/latest/databases/connecting#choose-when-metabase-syncs-and-scans

Or you can disable JSON unfolding in Admin > Databases > (db) > Show advanced options > disable "Unfold JSON columns", and click "Save changes", and then click "Sync database schema now".
https://www.metabase.com/docs/latest/databases/connections/postgresql#note-on-syncing-records-that-include-json

**Comment by tsmacdonald** [comment]
Presumably fixed by #22732
